leaderelection/leaderelection.go

```diff
@@ -141,10 +141,6 @@ func buildConfig(ctx context.Context, log *logp.Logger, client k8s.Interface, bl
 			},
 			OnStoppedLeading: func() {
 				log.Infof("leader election lock LOST, id: %v", id)
-				// leaderelection.Run stops in case it has stopped holding the leader lease, we re-run the manager
-				// to keep following leader status.
-				go manager.leader.Run(ctx)
-				log.Infof("re-running leader elector")
```

Comment:
I'm not super sure about this..
Removing this means we're missing the edge case of when cloudbeat is losing the leadership for some reason which is not container restart/cloudbeat reconfiguration (for example, network issues).
@uri-weisman wdyt exactly happened when this went crazy on your test?
were the leaderelector started multiple times?